### ISSUE TITLE
Configure support tools

### ISSUE THREAD

**Issue by Argmaster** [UniversityOfGdanskTeamPython/epseon_backend#5]
(Mostly) Python:
- pre-commit
- ruff
- poetry
- pytest
- pytest-cov
- codecov?
- prettier
- mkdocs

https://github.com/Argmaster/pygerber
https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks
https://python-poetry.org/docs/configuration/
https://docs.astral.sh/ruff/configuration/
https://pypi.org/classifiers/
https://semver.org/
https://docs.github.com/en/get-started/quickstart/github-flow
https://docs.github.com/en/actions

**Comment by Argmaster** [comment]
I have configured C++ as Python extension builds and CI for it. Additionally I have identified major problem with our approach to testing - we are creating GPU compute algorithm and we intended to use free Github Actions devices wich don't have GPUs :D
I'm currently trying to create some workaround for it by using siftshader - Vulkan implementation for CPU.

**Comment by Argmaster** [comment]
I had problems with getting shiftshader to work correctly. It took me some research on ruinning Vulkan applications without a display and multiple reattempts (27 to be precise) but I have managed to resolve it. It was caused by shiftshader shared objects being incorrectly copied via SCP. Afterwards I have managed to fully configure Python & CMake C++ builds with Vulkan API, pybind11, GoogleTest and other tools. Now they are successfully running on GitHub-provided machines.
